src/App.test.js

```diff
@@ -0,0 +1,41 @@
+import { screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { renderWithProviders } from "./test/utils"
+import App from './App'
+
+it('renders watch later link', () => {
```

Comment:
Refactoring the test suite to streamline the setup process is a great idea. You can improve maintainability by employing the beforeEach setup for rendering <App /> with providers.